### ISSUE TITLE
Some questions about the  reasoning output 

### ISSUE THREAD

**Issue by Doupi666** [zjunlp/KNN-KG#5]
Dear professor,
Thank you for reading my message.

I have read your paper, which name is Reasoning Through Memorization: Nearest Neighbor Knowledge Graph Embeddings, and I already run your code, and got result similar with yours. 

My question is about how to get the output like table 4 of the part 3.4 analysis. I do not know how to output entities with their probability like table 4 shown. 

Could your provide some ideas and method about my question?

If you could provide more specific information, I would be well grateful. Looking forward to your reply.

Kind regards,
Doupi

**Comment by Doupi666** [comment]
Would you mind open-sourcing the code which output the table 4?

**Comment by CheaSim** [comment]
In the Table 4, we randomly select some triples and get the probability of the output entity. You could write the code in the function `validation_step` in the `TransformerLitModel` to get the probability over the entities in KG and choose the top-5 entities. In this case we get the entity id, and you could use the entity2text.txt to get the real name of entity.

The code is very simple, you could try by yourself :)

**Comment by Doupi666** [comment]
I have tried to put the entity, relation as arguments in validation_step for geting some probability of the output entity, but it did not work. Actually I am not the major in code related field,  and still in self-study-step.  I want to experiment with one-to-many relational queries.Would you mind provide more detailed method?

**Comment by CheaSim** [comment]
https://github.com/zjunlp/KNN-KG/blob/70af5c71fce8743a6fe4809e29c69da109be3d45/lit_models/transformer.py#L105

You could get the probability over the entities in the `logits` after a `Softmax` function, and the entity id is the index of probability. I think it is quite simple as you only need to `print` entity id and find the entity name in the `entity2text.txt`.
